src/libraries/System.Text.RegularExpressions/tests/Regex.Match.Tests.cs

```diff
@@ -997,6 +997,17 @@ public void IsMatch_Invalid()
             Assert.Throws<ArgumentOutOfRangeException>(() => r.IsMatch("input", 6));
         }
 
+        [SkipOnTargetFramework(TargetFrameworkMonikers.NetFramework)] // take too long due to backtracking
+        [Theory]
+        [InlineData(@"(\w*)+\.", "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", false)]
```

Comment:
I think this is what I was thinking of when I opened the issue. 

Does this help a `(\w*)+` match succeed -- I assume that still runs interminably?

Reply:
> Does this help a (\w*)+ match succeed

Yes. This doesn't eliminate all backtracking, but it changes `(\w*)+` to `((?>\w*))+`, which means that it won't backtrack into the `\w+`.  It'll still do a lot of backtracking, but also a lot less than it otherwise would.

Note that it doesn't help with the original expression off of which you based this, the one commented out in the tests.  That one has catastrophic backtracking, and there's little we can do about that; try it out on regex101 and you'll see the same terribleness with pcre.